services/accumulo-utils/src/main/java/datawave/security/util/AuthorizationsMinimizer.java

```diff
@@ -0,0 +1,67 @@
+package datawave.security.util;
+
+import org.apache.accumulo.core.security.Authorizations;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.TreeSet;
+
+public class AuthorizationsMinimizer {
+    
+    public static Collection<Authorizations> minimize(Collection<Authorizations> authorizations) {
+        if (authorizations.size() > 1) {
+            // check if all auth sets are subsets of a minimum set, and use just the minimum if so
+            // minimize the sets to put all the common auths in the first one, and just the disjoint auths in the remainder.
+            LinkedHashSet<TreeSet<String>> allAuths = new LinkedHashSet<>();
+            for (Authorizations a : authorizations) {
+                TreeSet<String> s = new TreeSet<>();
+                for (byte[] b : a)
+                    s.add(new String(b));
+                allAuths.add(s);
+            }
+            
+            Collection<? extends Collection<String>> minimized = minimize2(allAuths);
+            authorizations = new LinkedHashSet<>(minimized.size());
+            for (Collection<String> a : minimized)
+                authorizations.add(new Authorizations(a.toArray(new String[a.size()])));
+        }
+        return authorizations;
+    }
+    
+    private static Collection<? extends Collection<String>> minimize2(Collection<? extends Collection<String>> authorizations) {
```

Comment:
maybe instead of minimize and minize2 we use apply and applyToStrings.  This can be done in a separate PR.